compiler/qsc_frontend/src/resolve.rs

```diff
@@ -593,14 +648,23 @@ impl With<'_> {
 
 impl AstVisitor<'_> for With<'_> {
     fn visit_namespace(&mut self, namespace: &ast::Namespace) {
-        let kind = ScopeKind::Namespace(Rc::clone(&namespace.name.name));
+        let ns = self
+            .resolver
+            .globals
+            .find_namespace(namespace.name.str_iter())
+            .expect("namespace should exist by this point");
+
+        let kind = ScopeKind::Namespace(ns);
         self.with_scope(namespace.span, kind, |visitor| {
+            // the below line ensures that this namespace opens itself, in case
+            // we are re-opening a namespace. This is important, as without this,
+            // a re-opened namespace would only have knowledge of its scopes.
+            visitor.resolver.bind_open(&namespace.name, &None);
```

Comment:
I don't follow - if this is important, how did we get by without it? Items in a namespace should already be available to the resolver.

FWIW, all the tests seem to pass if I comment out this line :P . If this is truly needed, maybe a unit test case demonstrating why?

Reply:
edited my above comment, you can ignore it.